frontend/src/js/upload-concept-list-modal/UploadConceptListModal.tsx

```diff
@@ -131,24 +376,84 @@ const UploadConceptListModal = ({ onAccept, onClose }: PropsT) => {
     }
   }, [filename]);
 
-  const selectConceptRootNodeAndResolveCode =
-    useSelectConceptRootNodeAndResolveCodes();
-
-  if (!conceptCodesFromFile || conceptCodesFromFile.length === 0) {
+  if (!fileRows || fileRows.length === 0) {
     onClose();
   }
 
-  const hasUnresolvedItems = unresolvedItemsCount > 0;
-  const hasResolvedItems = resolvedItemsCount > 0;
+  const onSubmit = useCallback(
```

Comment:
If you're not happy with this implementation, we could try moving the distinction of concepts/filters into the backend for the purpose of resolving?

Reply:
Hmm I know this looks a bit clunky, but I think it's actually okay, I will have to distinguish in the frontend in either case.